File: frontend/app/types/Portfolio.ts
// types/Post.ts

import { User } from './User';

export interface Image {
    ID: number;
    URL: string;
    PostID: number;
}

export interface Portfolio {
    ID: number;
    Title: string;
    Description: string;
    Skills: string[];
    Images: Image[];
    UserID: number;
    User: User;
}

File: backend/models/Portfolio.go
// models/post.go

package models

import (
	"github.com/lib/pq"
	"gorm.io/gorm"
)

type Post struct {
	gorm.Model
	Title       string `gorm:"not null"`
	Description string `gorm:"type:text"`
	GitHubLink  string
	ProductLink string
	Skills      pq.StringArray `gorm:"type:text[]"`
	Images      []Image        `gorm:"foreignKey:PostID"`
	UserID      uint           `gorm:"not null"`
	User        User           `gorm:"foreignKey:UserID"`
}

type Image struct {
	gorm.Model
	URL    string `gorm:"not null"`
	PostID uint   `gorm:"not null"`
}
